### PR TITLE
fix customized deviceshifu Content-Length Error

### LINKED ISSUE #510: Customized DeviceShifu Report an Error
**Describe the bug**

For the response's header is copied from the device's Response, but if costomized deviceShifu is used to process the Response, making it longer than before, it will cause errors in writing the Response.

**To Reproduce**

1. mock a http server
2. use costomized deviceShifu and make its return is longer than rawdata
3. call the api
4. deviceShifu will report an error

**Expected behavior**

report error 

**Log Output**
```
{"level":"error","ts":1677056467.9387946,"caller":"deviceshifuhttp/deviceshifuhttp.go:253","msg":"xxxx,error :http: wrote more than the declared Content-Length","stacktrace":"github.com/edgenesis/shifu/pkg/deviceshifu/deviceshifuhttp.DeviceCommandHandlerHTTP.commandHandleFunc.func1\n\t/shifu/pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go:253\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2109\nnet/http.(*ServeMux).ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2487\nnet/http.serverHandler.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2947\nnet/http.(*conn).serve\n\t/usr/local/go/src/net/http/server.go:1991"}
```

If applicable, add logs to help explain your problem.

**Server (please complete the following information):**

 - Shifu Version: v0.12.0
 - Kubernetes distro kind
 - Version v1.25.2
 - CPU Architecture arm64

**Additional context**

### PR DESCRIPTION
**Pull request by leepala**
**What this PR does / why we need it**:
fix customized deviceshifu Content-Length Error
**Will this PR make the community happier**? 
yes
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #510 

**How is this PR tested**
- [ ] unit test
- [ ] e2e test
- [x] other (please specify)
test handly
image

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go [hunk 1]
```diff
@@ -217,40 +217,37 @@ func (handler DeviceCommandHandlerHTTP) commandHandleFunc() http.HandlerFunc {
 			}
 		default:
 			http.Error(w, "not supported yet", http.StatusBadRequest)
-			logger.Errorf("Request type %v is not supported yet!", reqType)
+			logger.Errorf("Request type %sis not supported yet!", reqType)
```
Comment (BtXin) [inline pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go:220]: You still need a space here

File: pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go [hunk 2]
```diff
@@ -481,7 +478,7 @@ func (ds *DeviceShifuHTTP) collectHTTPTelemtries() (bool, error) {
 							respBodyString := utils.ProcessInstruction(deviceshifubase.PythonHandlersModuleName, instructionFuncName, rawRespBodyString, deviceshifubase.PythonScriptDir)
 							resp = &http.Response{
 								Header: make(http.Header),
-								Body: io.NopCloser(strings.NewReader(respBodyString)),
+								Body:   io.NopCloser(strings.NewReader(respBodyString)),
```
Comment (BtXin) [inline pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go:481]: Remove the space

**Review by BtXin — APPROVED** [review]
LGTM

**Review by tomqin93 — APPROVED** [review]
LGTM

**Pull request merged**